core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileObject.java

```diff
@@ -126,120 +134,74 @@ public ConnectionFileObject( AbstractFileName name, ConnectionFileSystem fs,
-    return resolvedFileObject.getRandomAccessContent( mode );
-  }
-
-  @Override public FileType getType() throws FileSystemException {
-    return resolvedFileObject.getType();
-  }
-
-  @Override public void holdObject( Object strongRef ) {
-    resolvedFileObject.holdObject( strongRef );
-  }
-
-  @Override public boolean isAttached() {
-    return resolvedFileObject.isAttached();
-  }
-
-  @Override public boolean isContentOpen() {
-    return resolvedFileObject.isContentOpen();
+  @Override
+  public FileContent getContent() throws FileSystemException {
+    return requireResolvedFileObject().getContent();
   }
 
-  @Override public boolean isExecutable() throws FileSystemException {
-    return resolvedFileObject.isExecutable();
+  @Override
+  public FileOperations getFileOperations() throws FileSystemException {
+    return requireResolvedFileObject().getFileOperations();
   }
 
-  @Override public boolean isFile() throws FileSystemException {
-    return resolvedFileObject.isFile();
+  @Override
+  public InputStream getInputStream() throws FileSystemException {
+    return requireResolvedFileObject().getInputStream();
   }
 
-  @Override public boolean isFolder() throws FileSystemException {
-    return resolvedFileObject.isFolder();
+  @Override
+  public OutputStream getOutputStream() throws FileSystemException {
+    return requireResolvedFileObject().getOutputStream();
   }
 
-  @Override public boolean isHidden() throws FileSystemException {
-    return resolvedFileObject.isHidden();
+  @Override
+  public OutputStream getOutputStream( boolean bAppend ) throws FileSystemException {
+    return requireResolvedFileObject().getOutputStream( bAppend );
   }
 
-  @Override public boolean isReadable() throws FileSystemException {
-    return resolvedFileObject.isReadable();
+  @Override
+  public RandomAccessContent getRandomAccessContent( RandomAccessMode mode ) throws FileSystemException {
+    return requireResolvedFileObject().getRandomAccessContent( mode );
   }
 
-  @Override public boolean isWriteable() throws FileSystemException {
-    return resolvedFileObject.isWriteable();
+  @Override
+  public void moveTo( FileObject destFile ) throws FileSystemException {
+    requireResolvedFileObject().moveTo( destFile );
   }
 
-  @Override public void moveTo( FileObject destFile ) throws FileSystemException {
-    resolvedFileObject.moveTo( destFile );
+  @Override
+  public void refresh() throws FileSystemException {
+    requireResolvedFileObject().refresh();
   }
 
-  @Override public void refresh() throws FileSystemException {
-    if ( resolvedFileObject != null ) {
-      resolvedFileObject.refresh();
-    }
-  }
+  @Nullable
+  public abstract FileObject getResolvedFileObject();
 
-  public FileObject getResolvedFileObject() {
-    return resolvedFileObject;
-  }
+  @NonNull
+  protected abstract AbstractFileObject<?> requireResolvedFileObject() throws FileSystemException;
```

Comment:
Allows subclasses to provide the most specific possible error message to satisfy https://hv-eng.atlassian.net/browse/BACKLOG-40732.